operator/webapp/core/k8s_client.py

```diff
@@ -0,0 +1,198 @@
+            status=500,
+            reason="Kubernetes cluster not reachable. Verify the cluster is running",
+        )
+
+    configmap_name = f"{route_data.route_name}-cm"
+    configmap = client.V1ConfigMap(
+        metadata=client.V1ObjectMeta(
+            name=configmap_name,
+            namespace=route_data.namespace,
+            labels={"app.kubernetes.io/created-by": "keip"},
+        ),
+        data={"integrationRoute.xml": route_data.route_file},
+    )
+
+    result = v1.list_namespaced_config_map(
+        namespace=route_data.namespace, field_selector=f"metadata.name={configmap_name}"
+    )
+
+    updated = False
+
+    if len(result.items) > 0:
+        # Update if exists
+        _LOGGER.info(
+            "Route ConfigMap '%s' already exists and will be updated", configmap_name
+        )
+
+        v1.replace_namespaced_config_map(
+            name=configmap_name, namespace=route_data.namespace, body=configmap
+        )
+        updated = True
+    else:
+        # Create if doesn't exist
+        _LOGGER.info(
+            "Route ConfigMap '%s' does not exist and will be created", configmap_name
+        )
+        configmap = v1.create_namespaced_config_map(
+            namespace=route_data.namespace, body=configmap
+        )
+
+    status = Status.UPDATED if updated else Status.CREATED
+    return Resource(status=status, name=configmap_name)
+
+
+def create_route_resources(route_data: RouteData) -> List[Resource]:
+    """
+    Creates both a ConfigMap and an Integration Route resource for the specified route configuration.
+
+    This function orchestrates the creation of two Kubernetes resources:
+    1. A ConfigMap containing the XML route payload for the integration route
+    2. An Integration Route resource that routes traffic based on the provided configuration
+
+    The function first creates the ConfigMap using the provided route data, then creates the Integration Route
+    using the ConfigMap name as the routeConfigMap reference. If a ConfigMap with the same name already exists,
+    it is updated rather than recreated. The Integration Route is created or updated based on the existing state.
+
+    Args:
+        route_data (RouteData): The route data containing the route name, namespace, and XML route file.
+            Must include all required fields to properly configure the integration route.
+
+    Returns:
+        List[Resource]: A list containing two Resource objects:
+            - The created/updated ConfigMap resource
+            - The created/updated Integration Route resource
+        The resources are returned in the order: [ConfigMap, Integration Route]
+
+    Raises:
+        ApiException: If the Kubernetes cluster is unreachable or if there is an error during API calls.
+        Exception: If an unexpected error occurs during processing or resource creation.
+    """
+    route_cm = create_route_configmap(route_data=route_data)
+    route = create_integration_route(
```

Comment:
:memo: nitpick - Seems `create_route_configmap` and `create_integration_route` are only called from tests. Should we add an underscore prefix to the function names to indicate they are 'internal' functions?